wire-runtime/src/main/java/com/squareup/wire/MessageAdapter.java

```diff
@@ -26,11 +26,14 @@
 import static com.squareup.wire.Preconditions.checkNotNull;
 
 public abstract class MessageAdapter<M> {
+  /** Returns the raw type of the messages this adapter can read and write. */
+  public abstract Class<?> messageType();
```

Comment:
Sadly that doesn't work for Lists. I could change it to return a `Type`, but that wouldn't be better.